composer/models/huggingface.py

```diff
@@ -74,6 +95,145 @@ def __init__(self,
+            local_checkpoint_save_location (Optional[Union[Path, str]], optional): If specified, where to save the checkpoint file to locally.
+                                                                                   If the input ``checkpoint_path`` is already a local path, this will be a symlink.
+                                                                                   Defaults to None, which will use a temporary file.
+
+        Raises:
+            ValueError: If the ``model_instantiation_class``, or the model class saved in the checkpoint, is not able to be imported
+
+        Returns:
+            Tuple[transformers.PreTrainedModel, Optional[transformers.PreTrainedTokenizer]]: The loaded HuggingFace model and (if present) tokenizer
+        """
+        try:
+            import transformers
+        except ImportError as e:
+            raise MissingConditionalImportError(extra_deps_group='transformers',
+                                                conda_package='transformers',
+                                                conda_channel='conda-forge') from e
+
+        # default local path to a tempfile if path is not provided
+        if local_checkpoint_save_location is None:
+            tmp_dir = tempfile.TemporaryDirectory()
+            local_checkpoint_save_location = Path(tmp_dir.name) / 'local-composer-checkpoint.pt'
+
+        if model_init_kwargs is None:
+            model_init_kwargs = {}
+
+        # download the checkpoint file
+        get_file(checkpoint_path, str(local_checkpoint_save_location))
+
+        # load the state dict in
+        loaded_state_dict = torch.load(local_checkpoint_save_location)
+
+        hf_state = loaded_state_dict['state']['integrations']['huggingface']
+        hf_model_state = hf_state['model']
+        hf_tokenizer_state = hf_state['tokenizer']
+
+        hf_config_dict = hf_model_state['config']['content']
+        # Update the config with any extra args needed
+        hf_config_dict.update(model_init_kwargs)
+        # JSON keys need to be converted back to ints, huggingface does not auto convert them along this code path
+        if 'id2label' in hf_config_dict:
+            hf_config_dict['id2label'] = {int(k): v for k, v in hf_config_dict['id2label'].items()}
+
+        loaded_config = transformers.AutoConfig.from_pretrained(hf_config_dict['_name_or_path'], **hf_config_dict)
+        if model_instantiation_class is not None:
+            # If the instantiation class is explicitly provided, use it
+            # The AutoModel* classes have `from_config`, while the PreTrainedModel classes do not
+            if issubclass(model_instantiation_class, transformers.models.auto.auto_factory._BaseAutoModelClass):
+                hf_model = model_instantiation_class.from_config(loaded_config)  # type: ignore
+            else:
+                hf_model = model_instantiation_class(loaded_config)
```

Comment:
Since we live and die by the YAML, is there a way to support string arguments for `model_instantiation_class` (i.e. if the input is a string, use that to import the class and then use the imported class)? Maybe this is already solved with Hydra, but we don't want to force that.

Reply:
yeah, should be pretty easy to add, will do.